js/ui/codeView.js

```diff
@@ -95,6 +96,24 @@ const _ensureHackDataFile = (function () {
         if (initialized)
             return keyfile;
 
+        for (let path of searchPaths) {
+            path = GLib.build_filenamev([path, 'hack-data.ini']);
+            const file = Gio.file_new_for_path(path);
+            const monitor = file.monitor_file(Gio.FileMonitorFlags.NONE, null);
+            monitor.connect('changed', (_monitor, _file, _otherfile, ev, _data) => {
+                if (ev === Gio.FileMonitorEvent.CHANGES_DONE_HINT) {
+                    // forces the keyfile reload the next time the function is called
+                    keyfile = new GLib.KeyFile();
+                    initialized = false;
+                    // disconnects all the previous monitors
```

Comment:
a. does it?
b. why is this desirable? Surely at the next call to the surrounding function, exactly the same set of monitors will be set up?

Reply:
Yes, you're right, we don't need to destroy/create again. I've updated the PR to create the monitors only the first time.